app/api/leaderboard/functions.ts

```diff
@@ -60,17 +62,54 @@ export const getLeaderboardData = async (
         return true;
       return false;
     })
+
     .sort((a, b) => {
+      if (
+        a.first_activity !== undefined &&
+        b.first_activity !== undefined &&
+        sortBy === "new_contributor"
+      ) {
+        return (
+          new Date(b.first_activity).getTime() -
+          new Date(a.first_activity).getTime()
+        );
+      }
       if (sortBy === "pr_stale") {
         return b.activityData.pr_stale - a.activityData.pr_stale;
       }
-      return b.summary[sortBy] - a.summary[sortBy];
+      if (sortBy !== "new_contributor") {
+        return (b.summary[sortBy] ?? 0) - (a.summary[sortBy] ?? 0);
+      }
+      return 0;
```

Comment:
You've already done an early return for the opposite of this condition, hence this check is not required

```suggestion
      return (b.summary[sortBy] ?? 0) - (a.summary[sortBy] ?? 0);
```